product_docs/docs/pgd/5/admin-tpa/upgrading_major_rolling.mdx

```diff
@@ -0,0 +1,537 @@
+!!!
+
+### Confirm the current Postgres version
+
+SSH into kaboom and use the following command to connect to PGD's `bdrdb` database:
+
+```bash
+sudo -u postgres psql bdrdb
+```
+
+When connected to `bdrdb`, use the following query:
+
+```sql
+SELECT * FROM bdr.group_versions_details;
+```
+
+The output is similar to this for your cluster:
+
+```
+  node_id   | node_name |        postgres_version        | bdr_version 
+------------+-----------+--------------------------------+-------------
+ 3490219809 | kaftan    | 15.5 (Debian 15.5-1EDB.buster) | 5.3.0
+ 2710197610 | kaboom    | 15.5 (Debian 15.5-1EDB.buster) | 5.3.0
+ 2111777360 | kaolin    | 15.5 (Debian 15.5-1EDB.buster) | 5.3.0
+(3 rows)
+```
+
+Confirm that the Postgres version is the expected version. Then, use the `exit` command to return to the node's command prompt.
+
+### Stop Postgres on the target node
+
+If you're not already on the target node, use SSH to connect to the target node (in this case, kaboom), and then run:
+
+```bash
+sudo systemctl stop postgres
+```
+
+This command halts the server on the current node, in this case, kaboom. Your cluster continues running using the other two nodes.
+
+### Install the new Postgres version
+
+While still on the target node, kaboom, install the new version of Postgres (PG16) and the upgrade tool:
+
+```bash
+sudo apt install edb-bdr5-pg16 edb-bdr-utilities
+```
+
+### Initialize Postgres 16
+
+Make a new data directory for the upgraded Postgres, and give the postgres user ownership of the directory:
+
+```bash
+sudo mkdir /opt/postgres/datanew
+sudo chown -R postgres:postgres /opt/postgres/datanew
+```
+
+Then, initialize Postgres 16 in the new directory:
+
+```bash
+sudo -u postgres /usr/lib/postgresql/16/bin/initdb \
+  -D /opt/postgres/datanew \
+  -E UTF8 \
+  --lc-collate=en_US.UTF-8 \
+  --lc-ctype=en_US.UTF-8 \
+  --data-checksums
+```
+
+This command creates a PG16 data directory for configuration, `/opt/postgres/datanew`.
+
+### Copy configuration for Postgres 15 to Postgres 16
+
+The next step copies the existing configuration (PG15) to the new PG16 data directory.
+
+Copy over the `postgresql.conf`, `postgresql.auto.conf`, and `pg_hba.conf` files and the whole `conf.d` directory:
+
+```bash
+sudo -u postgres cp /opt/postgres/data/postgresql.conf /opt/postgres/datanew/
+sudo -u postgres cp /opt/postgres/data/postgresql.auto.conf /opt/postgres/datanew/
+sudo -u postgres cp /opt/postgres/data/pg_hba.conf /opt/postgres/datanew/
+sudo -u postgres cp -r /opt/postgres/data/conf.d/ /opt/postgres/datanew/
+```
+
+### Rename the PGDATA directories
+
+Next, swap the PG15 and PG16 data directories: 
+
+```bash
+sudo mv /opt/postgres/data /opt/postgres/dataold
+sudo mv /opt/postgres/datanew /opt/postgres/data
+```
+
+!!! Important
```

Comment:
Important or a note. It's not the good path but something to refer to if it breaks.

Reply:
Note seems better I agree.